contracts/contract_caller.go

```diff
@@ -0,0 +1,67 @@
+package contracts
```

Comment:
sorry, i initially thought that this in the same file as registry.
do you plan to use it somewhere?

Reply:
@dshulyak  don't worry!
Yes this file should be passed to the registry because it implements the `abi/bind/backend/ContractCaller` interface and it will be instantiated with out `RCPClient`

Reply:
i suggest avoiding doing that. you can pass ethclient.NewClient and instantiate it using remote or local rpc client. regular rpc client, not our proxy

Reply:
@dshulyak why not using our `RPCClient`? otherwise if I use a normal RPCClient, where should I connect to then?

Reply:
you can connect either to local node (if there is one) or infura. it is also the same. it is just a matter of using original ethereum object instead of proxy. reason for doing so would be to avoid adding our own interface implementation when everything already exists

Reply:
> you can connect either to local node (if there is one) or infura
@dmitryn isn't that already handler by our `RPCClient`?

My idea was to call `NewContractCaller` with `StatusNode().RPCClient()`.

Or we can move this methods to our backend like we do for `SendTransaction`, so that `StatusBackend` can implement the `ContractCaller` interface.

Doing this we can just use the generated code passing `StatusBackend` to `NewRegistryCaller`.

What do you think?

Reply:
i think it will be good to reuse code implemented in ethclient.Client. you can hide it behind our backend if you want. my point was that reusing ethclient.Client allows to get rid from contract_caller.go file

Reply:
@dmitryn I know but `ethclient`wants an `rpc.Client` and not an interface so we can't use ours but it would be easier to it. If we pass a new `rpc.Client` we need a complete different client from the one we use for the reset of the transactions, or am I missing  something?

Reply:
this is simply a matter of passing client that is used by our client. because there is zero reasons to use our client, there is nothing in it that client can be useful for making a simple read from ethereum. if you dont want to use it ok